test/tests/components/content-service/content-service_test.go

```diff
@@ -0,0 +1,317 @@
+			InputOwnerID: "new-user",
+			InputName:    "test-blob",
+		},
+		{
+			Name:              "name with whitespace",
+			InputOwnerID:      gitpodBuiltinUserID,
+			InputName:         "whitespaces are not allowed",
+			ExpectedErrorCode: codes.InvalidArgument,
+		},
+		{
+			Name:              "name with invalid char",
+			InputOwnerID:      gitpodBuiltinUserID,
+			InputName:         "ä-is-not-allowed",
+			ExpectedErrorCode: codes.InvalidArgument,
+		},
+	}
+
+	uploadUrlRequest := features.New("UploadUrlRequest").
+		WithLabel("component", "content-service").
+		Setup(func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			api := integration.NewComponentAPI(ctx, cfg.Namespace(), cfg.Client())
+			return test_context.SetComponentAPI(ctx, api)
+		}).
+		Assess("it should run content-service tests", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			api := test_context.GetComponentAPI(ctx)
+
+			bs, err := api.BlobService()
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			for _, test := range tests {
+				t.Run(test.Name, func(t *testing.T) {
+					resp, err := bs.UploadUrl(ctx, &content_service_api.UploadUrlRequest{OwnerId: test.InputOwnerID, Name: test.InputName})
+					if err != nil && test.ExpectedErrorCode == codes.OK {
+						t.Fatal(err)
+						return
+					}
+					if err == nil && test.ExpectedErrorCode != codes.OK {
+						t.Fatalf("expected error with error code '%v' but got no error at all", test.ExpectedErrorCode)
+						return
+					}
+					if !hasErrorCode(err, test.ExpectedErrorCode) {
+						t.Fatalf("expected error with error code '%v' but got error %v", test.ExpectedErrorCode, err)
+						return
+					}
+					if err != nil && test.ExpectedErrorCode == codes.OK {
+						url := resp.Url
+						if url == "" {
+							t.Fatal("upload url is empty")
+						}
+						t.Logf("Got URL repsonse: %s", url)
+					}
+				})
+			}
+
+			return ctx
+		}).
+		Teardown(func(ctx context.Context, t *testing.T, _ *envconf.Config) context.Context {
+			api := test_context.GetComponentAPI(ctx)
+			defer api.Done(t)
+
+			return ctx
+		}).
+		Feature()
+
+	testEnv.Test(t, uploadUrlRequest)
+}
+
+func TestDownloadUrl(t *testing.T) {
+	tests := []struct {
+		Name              string
+		InputOwnerID      string
+		InputName         string
+		ExpectedErrorCode codes.Code
+	}{
+		{
+			Name:              "not existing download",
+			InputOwnerID:      gitpodBuiltinUserID,
+			InputName:         "this-does-not-exist",
+			ExpectedErrorCode: codes.NotFound,
+		},
+	}
+
+	downloadUrl := features.New("DownloadUrl").
+		WithLabel("component", "server").
+		Setup(func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			api := integration.NewComponentAPI(ctx, cfg.Namespace(), cfg.Client())
+			return test_context.SetComponentAPI(ctx, api)
+		}).
+		Assess("it should exists a builtin user workspace", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
```

Comment:
/nit 

"it should exists a builtin user workspace" => "it should get download url"?